### DISCUSSION TITLE
Gas Fee and MetaMask Notifications

### DISCUSSION THREAD

**Discussion by jaguard2021** [zkSync-Community-Hub/zksync-developers#276] [Dev Tooling] [answered]
I've successfully deployed a simple Hello World.sol contract on the ZKSync Mainnet using the Remix Plugin, with a remarkably low gas fee of only $1.23.

My question 

1. Is the gas fee cost for deploying a smart contract dependent on the code complexity within the smart contract?

2. Why is there no notification in MetaMask indicating the gas fee amount when deploying a smart contract transaction, unlike other types of transactions?

3. Why is there a lack of notification in the MetaMask activity column confirming that I have initiated a transaction to deploy a smart contract ?

Thanks dev team , finally I can create smart contracts in ZKsync via remix

**Comment by procebetjumorales** [comment]
ZKSync allows for deploying smart contracts with remarkably low gas fees, regardless of code complexity, making it cost-effective for users. 

 MetaMask may not provide gas fee notifications for smart contract deployments due to differences in transaction types and their associated metadata. 

 Lack of notification in the MetaMask activity column for smart contract deployments might be attributed to the absence of specific event triggers or metadata associated with contract creation transactions.

Happy smart contract deployment on ZKSync Mainnet via Remix!

**Comment by dgeyston** [comment]
1. Gas fees for deploying a smart contract can be influenced by code complexity, but they're primarily determined by factors like contract size and the operations performed during deployment.

2. MetaMask does not display gas fee amounts for smart contract deployment transactions because the fee estimation process for contract deployment is more complex and less predictable compared to regular transactions.

3. MetaMask may not provide a notification in the activity column for smart contract deployment transactions due to the asynchronous nature of contract deployment, which can result in delayed transaction confirmation messages.

**Comment by mariyokaveen** [comment]
Gas fees for deploying a smart contract can be influenced by factors like code complexity, contract size, and network congestion.

MetaMask may not display gas fee notifications for contract deployment transactions due to technical limitations or design choices.

The absence of a MetaMask activity notification for contract deployment transactions might be due to differences in how these transactions are processed compared to other types of transactions.

**Comment by bryanbrega** [comment]
Gas fees for deploying a smart contract are dependent on the code complexity. MetaMask does not display gas fees for contract deployment and lacks notifications for this type of transaction.

**Comment by bxpana** [comment, marked as answer]
@jaguard2021 
> 1. Is the gas fee cost for deploying a smart contract dependent on the code complexity within the smart contract?

Yes, the gas fee cost for deploying a smart contract is indeed dependent on the complexity of the contract. More complex operations require more computational resources and therefore require more gas to be executed.

> 2. Why is there no notification in MetaMask indicating the gas fee amount when deploying a smart contract transaction, unlike other types of transactions?

You mean when you deploy the contract via your connected wallet on Remix, Metamask doesn't show how much it's going to cost to deploy?

> 3. Why is there a lack of notification in the MetaMask activity column confirming that I have initiated a transaction to deploy a smart contract ?

Not sure on this one and might be a good question to ask MetaMask
